modules/cudf/src/node_cudf/table.hpp

```diff
@@ -172,11 +172,24 @@ class Table : public Napi::ObjectWrap<Table> {
     cudf::size_type threshold,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;
 
+  // table/copying.cpp
   ObjectUnwrap<Table> gather(
     Column const& gather_map,
     cudf::out_of_bounds_policy bounds_policy = cudf::out_of_bounds_policy::DONT_CHECK,
     rmm::mr::device_memory_resource* mr      = rmm::mr::get_current_device_resource()) const;
 
+  ObjectUnwrap<Table> scatter_scalar(
+    std::vector<std::reference_wrapper<const cudf::scalar>> const& source,
+    Column const& indices,
+    bool check_bounds                   = false,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;
+
+  ObjectUnwrap<Table> scatter_table(
+    Table const& source,
+    Column const& indices,
+    bool check_bounds                   = false,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;
+
```

Comment:
Seems like these could be two overloads of `scatter()`, no?

```suggestion
  ObjectUnwrap<Table> scatter(
    std::vector<std::reference_wrapper<const cudf::scalar>> const& source,
    Column const& indices,
    bool check_bounds                   = false,
    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;

  ObjectUnwrap<Table> scatter(
    Table const& source,
    Column const& indices,
    bool check_bounds                   = false,
    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource()) const;

```

Reply:
Unless you have strong feelings, I tend to prefer the separate functions to mirror the `libcudf` functions at the low-level (private-ish) `Table` level